app/commands/update_store_products.rb

```diff
@@ -9,6 +9,7 @@ def perform
         store_id: @store.id
       )
         product.attach_image_from_url(params.image)
+        product.set_average_color
```

Comment:
no se qué opinas pero lo del `set_average_color` me parece que debería ir en un callback del tipo `after_x` en product, para que siempre sea consistente. Me pasa que esta forma de hacerlo explícitamente hace que si después en otro lado quieres actualizar productos se pueda olvidar llamar al método.

Reply:
Si! traté de hacerlo con un callback, pero active storage no tiene callbacks al hacer un attachment :/

Reply:
Aah, te refieri a que no hay como `after_upload`? Porque me imagino que igual gatilla un update o un save al attachearle fotos, aunque no sea específico del update.

`before_update :set_average_color, if: -> { image.dirty? }` o algo por el estilo como `if: -> { image_changed? }`, no se si probaste eso también

Reply:
Hola! me colo a esta conversación :eyes:
Cuando se modifica el _attachment_ no se ejecutan los _callback_ tipo `after_upload` de los productos (https://stackoverflow.com/questions/53226228/callback-for-active-storage-file-upload)